### ISSUE TITLE
local -n variables may legitimately not be used after being assigned

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#3493]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034, SC2034, SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=bash
pai-restart-read () {
    local -n action=$1;shift
    local -n object=$1;shift

    read -r action object < "${pai_restart_file}"
}

declare left
declare right

pai-restart-read left right

echo "left is $left right is $right"
```

#### Here's what shellcheck currently says:
```
Line 6	SC2034: action appears unused. Verify use (or export if used externally).
Line 6	SC2034: object appears unused. Verify use (or export if used externally).
Line 6	SC2154: pai_restart_file is referenced but not assigned.
```
#### Here's what I wanted or expected to see:

Nothing. 'action' and 'object' are pass-by-reference variables. They are simply used to pass variables back to the caller. It's ok if unused in the function.
